### PR TITLE
Add agentic trace receipt summaries

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Add normalized sample-level receipt summary fields for replayed `agentic_tool_trace` samples.
- Preserve full `agentic_tool_observations` while exposing only receipt schema/count scalar metadata.
- Document the training trace summary contract and cover the replay path with changed-line coverage.

## Plan or Spec
- `docs/plans/2026-06-15-issue-1761-agentic-trace-receipt-summary.md`
- `docs/unified-agentic-tool-runtime-contract.md`

## Commands Run
```text
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx pytest -q services/mlx-worker-python/tests/test_training_dataset_builder.py services/mlx-worker-python/tests/test_pr_scoped_performance.py::test_scope_report_selects_training_dataset_probe
# 76 passed in 0.12s

.githooks/pre-commit
# pre-merge hook run completed on staged diff; performance report 20260615-144233-1051a42b: Status ok, 4 selected probes, 0 regressions, 0 verification failures, coverage 100.0%

git merge --no-edit origin/main
# merged origin/main at 39d2745c without conflicts

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx pytest -q services/mlx-worker-python/tests/test_training_dataset_builder.py services/mlx-worker-python/tests/test_pr_scoped_performance.py::test_scope_report_selects_training_dataset_probe
# 76 passed in 0.12s

make swift-test
# passed

make py-test
# 4033 passed, 14 skipped, 2 warnings in 167.20s

make integration-test
# 120 passed, 1 skipped in 456.37s

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx python <agentic summary path check>
# text_completion sample-limit path does not call the agentic receipt summary helper

UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx python <run_performance_report origin/main...HEAD>
# first merged run: Status regression, 1 sample-limit elapsed regression; investigated as unrelated micro-benchmark noise because the changed helper is not called by the text_completion sample-limit path
# second merged run: Status ok, 4 selected probes, 0 regressions, 0 verification failures, coverage 100.0%

python3 scripts/validate_pr_evidence.py --body-file .runtime/pr-body-issue-1761-receipt-summary.md
# PR evidence looks valid.

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx pytest services/mlx-worker-python/tests/test_training_dataset_builder.py
# 75 passed in 0.09s

git merge --no-edit origin/main
# merged origin/main at cb85e572 without conflicts

make swift-test
# passed

make py-test
# 4035 passed, 14 skipped, 2 warnings in 167.92s

make integration-test
# 120 passed, 1 skipped in 420.95s

UV_CACHE_DIR="$PWD/.uv-cache" UV_PYTHON=3.12 PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python --extra mlx python <run_performance_report origin/main...HEAD>
# Status ok, 4 selected probes, 0 regressions, 0 context regressions, 0 verification failures, changed-line coverage 100.0%

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx pytest services/mlx-worker-python/tests/test_training_dataset_builder.py::test_write_normalized_dataset_snapshot_applies_manifest_overrides -q
# red before helper fix: failed because no-schema receipts emitted "agentic_tool_untrusted_context_receipt_schema": ""
# green after helper fix: 1 passed in 0.06s

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" UV_PYTHON=3.12 uv run --project services/mlx-worker-python --extra mlx pytest -q services/mlx-worker-python/tests/test_training_dataset_builder.py services/mlx-worker-python/tests/test_pr_scoped_performance.py::test_scope_report_selects_training_dataset_probe
# 76 passed in 0.11s

.githooks/pre-commit
# review fix gate passed: make swift-test passed; make py-test 4035 passed, 14 skipped, 2 warnings in 167.38s; make integration-test 120 passed, 1 skipped in 410.61s; performance report Status ok, 4 selected probes, 0 regressions, 0 context regressions, 0 verification failures, changed-line coverage 100.0%
```

## Coverage and Metrics
- Changed-line coverage: `TOTAL 5 0 100%` for the review-fix changed lines in `training_dataset.py` and `test_training_dataset_builder.py`.
- Final local PR-scoped performance report: `.runtime/pre-commit-performance/20260615-163022-d438a223/report/report.md`.
- Final report summary: Status `ok`, changed files `4`, selected probes `4`, direct/gated probes `4`, regressions `0`, context regressions `0`, verification failures `0`.
- Probe coverage: all selected direct probes reported coverage `100.0%`.
- Observability mode: `N/A`; this change adds training-package metadata only and does not add runtime observability, sampled/debug/evidence-mode recording, or probe overhead.

## Known Gaps
- None.

## Evidence Checklist
- [x] Relevant plan or spec is identified.
- [x] Behavior changes are reflected in the relevant docs.
- [x] Protocol changes include regenerated generated artifacts. N/A: no protocol changes.
- [x] Dependency changes include updated lockfiles. N/A: no dependency changes.
- [x] Relevant tests were run.
- [x] A metrics report is included, or `N/A` is stated explicitly with the reason.
- [x] Observability mode, probe overhead, and any deferred debug or evidence work are recorded, or `N/A` is stated explicitly with the reason.
- [x] Deferred work and known gaps are stated explicitly.

### CODE REVIEW

File: services/mlx-worker-python/worker/model_ops/training_dataset.py [hunk 1]
```diff
@@ -1172,12 +1172,39 @@ def _agentic_trace_replay_turns(
...
+    return turns, replay_evidence
+
+
+def _agentic_tool_observation_receipt_summary(
+    observations: Iterable[Mapping[str, Any]],
+) -> dict[str, Any]:
+    receipt_schema = ""
+    receipt_count = 0
+    for observation in observations:
+        receipts = observation.get("untrusted_context_receipts")
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply services/mlx-worker-python/worker/model_ops/training_dataset.py:1194]: Fixed. `_agentic_tool_observation_receipt_summary` now skips non-mapping observations before reading receipt fields, and the test covers a `None` observation input.

File: services/mlx-worker-python/worker/model_ops/training_dataset.py [hunk 2]
```diff
@@ -1172,12 +1172,41 @@ def _agentic_trace_replay_turns(
...
+            if not isinstance(receipt, Mapping):
+                continue
+            receipt_count += 1
+            schema_version = receipt.get("schema_version")
+            if not receipt_schema and isinstance(schema_version, str):
+                receipt_schema = schema_version
+    if receipt_count == 0:
+        return {}
+    return {
+        "agentic_tool_untrusted_context_receipt_schema": receipt_schema,
```
Comment (Zigfreidish) [inline services/mlx-worker-python/worker/model_ops/training_dataset.py]: Edge case: if mapping-shaped receipts exist but none carry a string `schema_version`, `receipt_schema` stays `""` and this line returns `"agentic_tool_untrusted_context_receipt_schema": ""`. Downstream readers that do a truthiness check on the schema field will treat it as absent even though receipts were counted. The existing tests always have at least one receipt with a `schema_version`, so this path is untested. Consider either omitting the schema key when it stays empty, or adding a test for the no-schema-version case so the empty-string behavior is explicit and intentional.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by Zigfreidish — COMMENTED** [review]
**Review: LGTM**

Well-scoped addition. `_agentic_tool_observation_receipt_summary` is clean — it counts mapping-shaped receipts and extracts only the first string schema version without copying any payload content. The contract doc update in `unified-agentic-tool-runtime-contract.md` correctly matches the implementation. Full test suite passing, coverage 100%, performance status `ok`. One inline note about an untested edge case (receipts exist but none have a schema_version → empty string in the schema field) — worth a follow-up test or an explicit doc note, not a blocker.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: services/mlx-worker-python/worker/model_ops/training_dataset.py (hunk 2 above)
Comment (Zigfreidish) [reply services/mlx-worker-python/worker/model_ops/training_dataset.py]: Fixed. The receipt summary now keeps `agentic_tool_untrusted_context_receipt_count` when mapping-shaped receipts exist, and only emits `agentic_tool_untrusted_context_receipt_schema` when a counted receipt has a string `schema_version`. The test covers the no-schema receipt case, and the contract/plan describe the conditional schema field.
